### PR TITLE
Updated gradient descent and some minor changes done

### PR DESCRIPTION
**Pull request by kidrahahjo**

### CODE REVIEW

**Comment by antmarakis** [comment]
Hello, I am afraid you have changed/added too many files, I cannot merge this as is even though it is nice work :/

**Comment by kidrahahjo** [comment]
Hello thanks for your reply,
I'll look into it for sure. So am I supposed to take my commits back and
perform them one by one?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "venv/bin/python2.7"
```
Comment (antmarakis) [inline .vscode/settings.json]: Thing is, you have created a lot of files that are unnecessary, like this one. Your only changes seem to be in learning and text, but you have created around 50 more files.

**Comment by kidrahahjo** [comment]
Yes, I've corrected these things locally.
Due to my lectures I was unable to push it.
I'll push them by today itself.

**Pull request closed without merging**